src/shared/ui/Input.tsx

```diff
@@ -1,20 +1,23 @@
 import {FieldPath, UseFormRegisterReturn} from "react-hook-form";
 import {useState} from "react";
-import {ILoginFormValues, IRegisterFormValues} from "@/shared/types";
+import {IGlobalFilterFormValues, ILoginFormValues, IRegisterFormValues} from "@/shared/types";
 
 interface IInputProps {
   onChange?: () => void;
   HTMLType: 'email' | 'text' | 'password';
   iconType?: 'person' | 'mail' | 'password' | 'search-icon-gray';
   value?: string;
   placeholder?: string;
-  extraClass?: string;
+  extraBoxClass?: string;
+  extraInputClass?: string;
   defaultValue?: string;
   error?: any;
-  hookFormValues?: UseFormRegisterReturn<FieldPath<ILoginFormValues>> | UseFormRegisterReturn<FieldPath<IRegisterFormValues>>;
+  hookFormValues?: UseFormRegisterReturn<FieldPath<ILoginFormValues>> | UseFormRegisterReturn<FieldPath<IRegisterFormValues>> | UseFormRegisterReturn<FieldPath<IGlobalFilterFormValues>>;
```

Comment:
@snurnisyan Isn't there a more convenient way to specify this type than just building up the possible types via or?

Reply:
I think we can try something like this 

```suggestion
  hookFormValues?: UseFormRegisterReturn<string>
```

because there soon will be lots of different form names.